src/hoc/RealTime.tsx

```diff
@@ -34,58 +32,37 @@ export const RealTime = ({ children }: { children: ReactNode }) => {
       const updatedTask = payload.new
       //check if the new task in this event belongs to the same workspaceId
       if (payload.new.workspaceId === tokenPayload?.workspaceId) {
+        //if the task is deleted
         if (updatedTask.deletedAt) {
           const newTaskArr = tasks.filter((el) => el.id !== updatedTask.id)
           store.dispatch(setTasks(newTaskArr))
+          //if a user is in the details page when the task is deleted then we want the user to get redirected to '/' route
+          if (pathname.includes('detail')) {
+            if (pathname.includes('cu')) {
+              router.push(`/client?token=${token}`)
+            } else {
+              router.push(`/?token=${token}`)
+            }
+          }
+          //if the task is updated
         } else {
           const newTaskArr = [...tasks.filter((task) => task.id !== updatedTask.id), updatedTask]
           store.dispatch(setTasks(newTaskArr))
         }
       }
     }
   }
-  const handleTaskRealTimeUpdatesDetails = (payload: RealtimePostgresChangesPayload<RealTimeTaskResponse>) => {
-    const updatedTask = payload.new as RealTimeTaskResponse
-    if (updatedTask.deletedAt) {
-      if (path.includes('/iu')) {
-        router.push(`/?token=${token}`)
-      } else {
-        router.push(`/client?token=${token}`)
-      }
-    } else {
-      store.dispatch(setTask(updatedTask))
-    }
-  }
 
   useEffect(() => {
-    if (path.includes('/detail') && task) {
-      const channel = supabase
-        .channel('realtime tasks details')
-        .on(
-          'postgres_changes',
-          {
-            event: '*',
-            schema: 'public',
-            table: 'Tasks',
-            filter: `id=eq.${task.id}`,
-          },
-          handleTaskRealTimeUpdatesDetails,
-        )
-        .subscribe()
-      return () => {
-        supabase.removeChannel(channel)
-      }
-    } else {
-      const channel = supabase
-        .channel('realtime tasks')
-        .on('postgres_changes', { event: '*', schema: 'public', table: 'Tasks' }, handleTaskRealTimeUpdatesInRoot)
-        .subscribe()
+    const channel = supabase
+      .channel('realtime tasks')
+      .on('postgres_changes', { event: '*', schema: 'public', table: 'Tasks' }, handleTaskRealTimeUpdates)
```

Comment:
This is a major security concern because we are getting CRUD events for all tasks, not just tasks that IU has access to. 

E.g. if I had two companies Apple and Samsung, Samsung will be subscribed to changes for tasks in Apple as well.

We should at least add filters for workspaceId.

```js
.on('postgres_changes', { event: '*', schema: 'public', table: 'Tasks', filter: `workspaceId=eq.${tokenPayload.workspaceId}` }, handleTaskRealTimeUpdates)
```

Other required filters should be:
- Adding a filter for current user's companyAccessList if user is IU (using `in.` filter)
- Only subscribing tasks where `assigneeId=eq.${tokenPayload.clientId}` for client users.

These later ones can be added and dealt with in later tickets since it requires some additional logic for isClientAccessRestricted, etc, and we are in a time crunch right now.

Reply:
I think this thing can be approached separately after the release because there are a lot of open holes for optimizations which may block the release. May be creating a separate ticket would be better.

Reply:
Could you still please add the workspaceId filter

Reply:
Other filters can wait but this is a major one, especially since the code is open sourced